apps/minio/deployers/test.yml

```diff
@@ -0,0 +1,142 @@
+              k8s_facts:
+                kind: StorageClass
+                name: "{{ lookup('env','PROVIDER_STORAGE_CLASS') }}"
+              register: result
+              failed_when: result.resources | length < 1
+
+              # Creating namespace
+            - include_tasks: /utils/k8s/create_ns.yml
+
+            - name: Get the application label values from env
+              set_fact:
+                app_lkey: "{{ app_label.split('=')[0] }}"
+                app_lvalue: "{{ app_label.split('=')[1] }}"
+
+            - name: Replace the application label placeholder in service spec
+              replace:
+                path: "{{ pvc_deployment }}"
+                regexp: "lkey: lvalue"
+                replace: "{{ app_lkey }}: {{ app_lvalue }}"
+
+            - name: Replace the storage capacity placeholder
+              replace:
+                  path: "{{ pvc_deployment }}"
+                  regexp: "volume-capacity"
+                  replace: "{{ lookup('env','CAPACITY') }}" 
+
+            - name: Replace the storage class placeholder
+              replace:
+                  path: "{{ pvc_deployment }}"
+                  regexp: "testclass"
+                  replace: "{{ lookup('env','PROVIDER_STORAGE_CLASS') }}" 
+
+            - name: Replace the PVC name placeholder
+              replace:
+                  path: "{{ pvc_deployment }}"
+                  regexp: "testclaim"
+                  replace: "{{ lookup('env','APP_PVC') }}"
+
+            - name: Applying the pvc
+              shell: kubectl apply -f minio-pvc.yml -n {{ app_ns }}
+            
+            - name: Checking the status of PVC
+              shell: kubectl get pvc {{ app_pvc }} -n {{ app_ns }} -o jsonpath='{.status.phase}'
+              register: pvc_status
+              until: "'Bound' in pvc_status.stdout"
+              delay: 30
+              retries: 15
+
+          when: lookup('env','ACTION') == 'provision'      
+      
+      ## create and delete the deployments according to action
+        - include_tasks: ./minio_prerequisite.yml
+          loop: "{{ range(0, deploy_count | int, 1)|list }}"
+
+        - block:
+            ## verify and delete the pvc 
+            - name: delete the pvc
+              shell: >
+                kubectl delete pvc {{ app_pvc }} -n {{ app_ns }} 
+              args:
+                executable: /bin/bash
+              register: status
+              failed_when: "status.rc != 0"
+
+            ## verify if the pvc is deleted
+            - name: check if the cloned pvc is deleted
+              shell: >
+                kubectl get pvc -n {{ app_ns }}
+              args:
+                executable: /bin/bash
+              register: pvc_name
+              until: "app_pvc not in pvc_name.stdout"
+              delay: 30
+              retries: 15   
+            
+            - name: Delete the service
+              shell: kubectl delete service minio-service -n {{ app_ns }}
```

Comment:
This task is not needed. Deletion of Service has been taken care as part of pvc deletion

Reply:
resolved